markdown.dtx

```diff
@@ -15611,6 +15705,27 @@ function M.writer.new(options)
     local name = util.cache(options.cacheDir, s, nil, nil, ".verbatim")
     return {"\\markdownRendererInputFencedCode{",name,"}{",i,"}"}
   end
+  function self.jekyllData(_, data)
```

Comment:
Why does the `writer->jekyllData` method receive two parameters? If possible, we would like to keep the interface clean. Additionally, the `writer->jekyllData` method should be documented like the rest of the `writer` methods. What is the structure of `data`?

Reply:
I constructed it similar to the fenced code constructs, they had two parameters I can strip one by using a function when calling to strip the first. Maybe that's easier. 

The data is a table containing the input lines of the whole jekyll block.

Reply:
> I constructed it similar to the fenced code constructs, they had two parameters I can strip one by using a function when calling to strip the first. Maybe that's easier.

Perhaps we don't need to produce the first parameter at all? Fenced code blocks accept an infostring to the right of ```` 
``` ````, whereas I assume we just want YAML metadata to start with a `---` (although it is true that [a compliant YAML document may contain text to the right of `---`](https://yaml.org/spec/1.2/#id2761719)).

Reply:
I now drop `infostring` by the inline function before calling `writer.jekyllData`.